docs/source/jit.rst

```diff
@@ -100,6 +100,129 @@ Example::
 
     my_script_module = torch.jit.script(MyScriptModule())
 
+Migrating to PyTorch 1.2 Recursive Scripting API
+------------------------------------------------
+This section details the changes to TorchScript in PyTorch 1.2. If you are new to TorchScript you can
+skip this section.
+
+Overview
+~~~~~~~~
+PyTorch 1.2 brings changes to the scripting API to close the usability gap between the TorchScript compiler and the tracer. Rather than inherit directly from ``torch.jit.ScriptModule``, a ``torch.nn.Module`` instance can be passed to ``torch.jit.script()``, which returns a compiled ``ScriptModule`` version of the module.
+  * The module's ``forward`` is compiled by default. Methods called from ``forward`` are lazily compiled in the order they are used in ``forward``.
```

Comment:
I think it would be clearer if there was a "this is what happens when a module is passed to `torch.jit.script()`". Or are you planning to write this for the general docs?

Reply:
Should we move the full explanation of what happens in `script()` in the docstring for `script()` and drop it from here?